src/base/torrentfileswatcher.cpp

```diff
@@ -159,7 +159,7 @@ void TorrentFilesWatcher::initWorker()
     connect(m_asyncWorker, &TorrentFilesWatcher::Worker::torrentFound, this, &TorrentFilesWatcher::onTorrentFound);
 
     m_asyncWorker->moveToThread(m_ioThread.get());
-    connect(m_ioThread.get(), &QObject::destroyed, this, [this] { delete m_asyncWorker; });
+    connect(m_ioThread.get(), &QThread::finished, m_asyncWorker, [this] { delete m_asyncWorker; });
```

Comment:
will this work?
```suggestion
    connect(m_ioThread.get(), &QThread::finished, m_asyncWorker, &QObject::deleteLater);
```

Reply:
> will this work?

No. Since event loop of finished thread is stopped.


> this pattern is used frequently in the code base.

It can be used to delete object in the thread other than finished one. This is not appropriate in this case.

Reply:
>It can be used to delete object in the thread other than finished one. This is not appropriate in this case.

I don't understand, care to explain more? The Qt docs has an example and it seems quite similar to our case here: https://doc.qt.io/qt-6/qthread.html#details

Off topic, in the worker class the member variables can switch from pointers to normal variables in order to save another layer of indirection, or there are other benefits for using pointers here?

Reply:
> (Just for discussion, not blocking merging)
> 
> > It can be used to delete object in the thread other than finished one. This is not appropriate in this case.

It looks like I misled you. I thought in a hurry that you were suggesting the following:
```c++
connect(m_ioThread.get(), &QThread::finished, this, [this] { m_asyncWorker->deleteLater(); });
```
So my answer referred to this option. As for your real suggestion:
```c++
connect(m_ioThread.get(), &QThread::finished, m_asyncWorker, &QObject::deleteLater);
```
Yes, this is the commonly used method. And it was so initially. But it caused the crash of the application when exiting since Qt 6.5 (on some platforms and build variants). So I changed it in #19169 with the following:
```c++
connect(m_ioThread.get(), &QThread::finished, this, [this] { delete m_asyncWorker; });
```
As it turned out, it had another problem - a memory leak, although not dangerous in practice, but I decided to fix it anyway (because for many users, a "memory leak" is like a red rag for a bull, although most of them do not understand what it really is). I changed it with current version:
```c++
connect(m_ioThread.get(), &QObject::destroyed, this, [this] { delete m_asyncWorker; });
```
But it has `QObject::~QObject: Timers cannot be stopped from another thread`.